pkgs/by-name/pe/perl-debug-adapter/package.nix

```diff
@@ -0,0 +1,47 @@
+{ lib
+, buildNpmPackage
+, fetchFromGitHub
+, fetchpatch
+, makeWrapper
+, perl
+# Needed if you want to use it for a perl script with dependencies.
+, extraPerlPackages ? []
```

Comment:
Wouldn't it be easier to use, if this would be impure?

Reply:
> Wouldn't it be easier to use, if this would be impure?

I was thinking about it, but it is a bit tricky to notice the `PadWalker` dependency is needed - hence the default derivation at least will work for Perl scripts without additional external dependencies. If a Nix user wants to use this LSP server with a Perl script that includes more dependencies, they can do that rather easily, and I assume they already have a `shell.nix` setup or a flake setup to inject this override in if needed. I think also the `$PERL5LIB` environment variable will still work in this case.

Reply:
If that works that would be great and I think a bit more preferred than this option but fine for me.